### ISSUE TITLE
[java] PublicMemberInNonPublicType can only be suppressed on type and not member level

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#6461]
> BTW: In order to ignore the violation I need to suppress on the type and not on the affected method, it would be nice if that can be changed as well.  

 _Originally posted by @uhafner in [#6459](https://github.com/pmd/pmd/issues/6459#issuecomment-3904187161)_

**Comment by UncleOwen** [comment]
Cannot reproduce. This code ...

```java
class MyTest {
    // @SuppressWarnings("PMD.PublicMemberInNonPublicType")
    public static void main(final String... args)  {
        // main method that is invoked in IntelliJ
    }
}
```

... does trigger a `PublicMemberInNonPublicType` violation in both 7.22.0 and 7.25.0. The violation goes away if I remove the `//` in line 2.

@uhafner Could you please double-check?

**Comment by adangel** [comment]
@UncleOwen In any case, I think it makes sense, to add such an example as a test case. To be sure, we can actually suppress the violation.

**Comment by UncleOwen** [comment]
@adangel agreed.
